integration/pkcs11/pkcs11_test.go

```
@@ -0,0 +1,420 @@
+	caBytes, err := ioutil.ReadFile(filepath.Join(orgCAPath, fmt.Sprintf("ca.%s.example.com-cert.pem", orgName)))
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Updating the peer signcerts")
+
+	newOrdererPemCert := buildCert(network, caBytes, orgCAPath, peerCSR, peerSerial, peerPubKey)
+
+	updateMSPFolder(network.PeerLocalMSPDir(peer), fmt.Sprintf("peer.%s.example.com-cert.pem", orgName), newOrdererPemCert)
+
+	By("Updating the peer admin user signcerts")
+	newAdminPemCert := buildCert(network, caBytes, orgCAPath, adminCSR, adminSerial, adminPubKey)
+
+	orgAdminMSPPath := network.PeerUserMSPDir(peer, "Admin")
+	updateMSPFolder(orgAdminMSPPath, fmt.Sprintf("Admin@%s.example.com-cert.pem", orgName), newAdminPemCert)
+
+}
+
+func configureOrdererPKCS11(ctx *pkcs11.Ctx, sess pkcs11.SessionHandle, network *nwo.Network) {
+
+	orderer := network.Orderer("orderer")
+	orgName := "orderer"
+
+	ordererPubKey, ordererCSR, ordererSerial := createCSR(ctx, sess, orgName, "orderer")
+	adminPubKey, adminCSR, adminSerial := createCSR(ctx, sess, orgName, "admin")
+
+	// Retrieves org CA cert
+	orgCAPath := network.OrdererOrgCADir()
+	caBytes, err := ioutil.ReadFile(filepath.Join(orgCAPath, "ca.example.com-cert.pem"))
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Updating the orderer signcerts")
+
+	newOrdererPemCert := buildCert(network, caBytes, orgCAPath, ordererCSR, ordererSerial, ordererPubKey)
+
+	updateMSPFolder(network.OrdererLocalMSPDir(orderer), "orderer.example.com-cert.pem", newOrdererPemCert)
+
+	By("Updating the orderer admin user signcerts")
+	newAdminPemCert := buildCert(network, caBytes, orgCAPath, adminCSR, adminSerial, adminPubKey)
+
+	orgAdminMSPPath := network.OrdererUserMSPDir(orderer, "Admin")
+	updateMSPFolder(orgAdminMSPPath, "Admin@example.com-cert.pem", newAdminPemCert)
+
+}
+
+// Creates pkcs11 context and session
+func setupPKCS11Ctx(lib, label, pin string) (*pkcs11.Ctx, pkcs11.SessionHandle, error) {
```

Why do we need to return an error? You're already asserting on errors.